packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -65,30 +66,21 @@ abstract class OverlayRoute<T> extends Route<T> {
   @override
   void dispose() {
     finished();
+    super.dispose();
   }
 }
 
 /// A route with entrance and exit transitions.
 abstract class TransitionRoute<T> extends OverlayRoute<T> {
   /// Creates a route with entrance and exit transitions.
   TransitionRoute({
-    Completer<T> popCompleter,
     Completer<T> transitionCompleter
-  }) : _popCompleter = popCompleter,
-       _transitionCompleter = transitionCompleter;
+  }) : _transitionCompleter = transitionCompleter;
 
   /// The same as the default constructor but callable with mixins.
   TransitionRoute.explicit(
-    Completer<T> popCompleter,
     Completer<T> transitionCompleter
-  ) : this(popCompleter: popCompleter, transitionCompleter: transitionCompleter);
-
-  /// This future completes once the animation has been dismissed. For
-  /// ModalRoutes, this will be after the completer that's passed in, since that
-  /// one completes before the animation even starts, as soon as the route is
-  /// popped.
```

Comment:
we should mention these details (from the removed comment) maybe in the `completed` docs or something